zk.js/src/utils.ts

```diff
@@ -39,11 +39,32 @@ import * as os from "os";
 const crypto = require("@noble/hashes/crypto");
 
 export function hashAndTruncateToCircuit(data: Uint8Array) {
-  return new BN(
-    sha256.create().update(Buffer.from(data)).digest().slice(1, 32),
-    undefined,
-    "be",
-  );
+  return truncateToCircuit(sha256.create().update(Buffer.from(data)).digest());
+}
+
+/**
+ * Truncates the given 32-byte array to a 31-byte one, ensuring it fits
+ * into the Fr modulo field.
+ *
+ * ## Safety
+ *
+ * This function is primarily used for truncating hashes (e.g., SHA-256) which are
+ * not constrained by any modulo space. It's important to note that, as of now,
+ * it's not possible to use any ZK-friendly function within a single transaction.
+ * While truncating hashes to 31 bytes is generally safe, you should ensure that
+ * this operation is appropriate for your specific use case.
+ *
+ * @param bytes The 32-byte array to be truncated.
+ * @returns The truncated 31-byte array.
+ *
+ * @example
+ * ```typescript
+ * // example usage of truncate function
+ * const truncated = truncateFunction(original32BytesArray);
+ * ```
+ */
+export function truncateToCircuit(digest: Uint8Array) {
```

Comment:
I think it's safer to always hash and truncate. In the case of the integrity hash you are also just [hashing here](https://github.com/Lightprotocol/light-protocol/pull/299/files#diff-a4ab527566fa152c98d7c43f4873cc8676154ab7f9f5f68d64d25729265b48daR402-R423) and then truncate.